### PR TITLE
Replace golang/mock to uber-go/mock.

### PR DESCRIPTION
**Pull request by liangyuanpeng**
golang/mock is no longer maintained and should replace to go.uber.org/mock, check https://github.com/golang/mock

### CODE REVIEW

File: proto/agent/mocks/agent_mock.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	context "context"
 	reflect "reflect"
 
-	gomock "github.com/golang/mock/gomock"
+	gomock "go.uber.org/mock/gomock"
```
Comment (jkh52) [inline proto/agent/mocks/agent_mock.go:26]: Did you run `make gen` or `make mock_gen` to generate this? What version of mockgen did you install?

I haven't tried it, but if appropriate update the mockgen reference (https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/README.md#mockgen) to https://github.com/uber-go/mock

**Comment by liangyuanpeng** [comment]
Test success at github action.
```shell
go test -race -covermode=atomic -coverprofile=konnectivity.out ./... && go tool cover -html=konnectivity.out -o=konnectivity.html
```

/test pull-apiserver-network-proxy-test

**Pull request merged**